src/Parser/Mage/Frost/Modules/Traits/WintersReach.js

```diff
@@ -0,0 +1,157 @@
+import SpellLink from 'common/SpellLink';
+import SpellIcon from 'common/SpellIcon';
+import Analyzer from 'Parser/Core/Analyzer';
+import StatisticBox, { STATISTIC_ORDER } from 'Interface/Others/StatisticBox';
+import { formatMilliseconds, formatNumber, formatPercentage } from 'common/format';
+
+const CAST_BUFFER = 250;
+
+const debug = false;
+
+class WintersReach extends Analyzer {
+
+  beginCastTimestamp = 0;
+  castTimestamp = 0;
+  buffUsed = false;
+  beginCastFound = false;
+  count = 0;
+  wastedProcs = 0;
+  usedProcs = 0;
+  totalProcs = 0;
+
+  constructor(...args) {
+    super(...args);
+    this.active = this.selectedCombatant.hasTrait(SPELLS.WINTERS_REACH_TRAIT.id);
+  }
+
+  on_byPlayer_applybuff(event) {
+    const spellId = event.ability.guid;
+    if (spellId !== SPELLS.WINTERS_REACH_BUFF.id) {
+      return;
+    }
+    this.buffUsed = false;
+    this.totalProcs += 1;
+    this.buffAppliedTimestamp = event.timestamp;
+    debug && console.log("Buff Applied @ " + formatMilliseconds(event.timestamp - this.owner.fight.start_time));
+  }
+
+  on_byPlayer_refreshbuff(event) {
+    const spellId = event.ability.guid;
+    if (spellId !== SPELLS.WINTERS_REACH_BUFF.id) {
+      return;
+    }
+    this.buffUsed = false;
+    this.wastedProcs += 1;
+    this.totalProcs += 1;
+    debug && console.log("Buff Refreshed " + formatMilliseconds(event.timestamp - this.owner.fight.start_time));
+  }
+
+  on_byPlayer_removebuff(event) {
+    const spellId = event.ability.guid;
+    if (spellId !== SPELLS.WINTERS_REACH_BUFF.id) {
+      return;
+    }
+    debug && console.log("Buff Removed @ " + formatMilliseconds(event.timestamp - this.owner.fight.start_time));
+    if (this.buffUsed === false) {
+      this.wastedProcs += 1;
+      debug && console.log("Buff Expired @ " + formatMilliseconds(event.timestamp - this.owner.fight.start_time));
+    } else {
+      this.usedProcs += 1;
+    }
+  }
+
+  on_byPlayer_begincast(event) {
+    const spellId = event.ability.guid;
+    if (spellId !== SPELLS.FLURRY.id) {
+      return;
+    }
+    this.beginCastTimestamp = event.timestamp;
+    this.beginCastFound = true;
+
+    debug && console.log("Flurry Begin Cast @ " + formatMilliseconds(event.timestamp - this.owner.fight.start_time));
+  }
+
+  on_byPlayer_cast(event) {
+    const spellId = event.ability.guid;
+    if (spellId !== SPELLS.FLURRY.id || this.beginCastFound === false) {
+      return;
+    }
+    this.beginCastFound = false;
+    debug && console.log("Flurry Casted @ " + formatMilliseconds(event.timestamp - this.owner.fight.start_time));
+    this.castTimestamp = event.timestamp;
+    const castTime = this.castTimestamp - this.beginCastTimestamp;
+    //Checks the begincast and cast timestamps to determine if it is instant cast or not. This doesnt matter for ABT and ToS because winters reach flurries dont have a begincast, but in Nighthold they do. So this needs to remain for backwards compatibility
```

Comment:
You don't have to make it compatible for Legion patches, particularly in this case since you couldn't have the trait in the Nighthold patch anyway.

Reply:
Yea, I just copied/pasted Pyroclasm from fire since its pretty much the exact same thing. Ill go through it again and make sure all of those suggestions are taken care of.